lib/tasks_private/terraform/provision.rake

```diff
@@ -0,0 +1,84 @@
+require "json"
+require 'yaml'
+require 'logger'
+require 'ruby-terraform'
+
+namespace :provision do
+  desc "check terraform is installed"
+  task :check => :environment do
+    system("command -v terraform > /dev/null") or raise("install terraform to provision powervc vm")
+  end
+
+  desc "provision a vm"
+  task :apply => :check do
+    cdir = ENV['cwd'] || Dir.pwd
+    tdir = __dir__
+    config = setup(cdir, tdir)
+    plan(config)
+    apply(cdir, tdir)
+  end
+
+  desc "delete a vm"
+  task :destroy => :check do
+    cdir = ENV['cwd'] || Dir.pwd
+    tdir = __dir__
+    config = setup(cdir, tdir)
+    destroy(cdir, config)
+  end
+
+  def setup(cdir, tdir)
+    config = YAML.safe_load(File.read(File.join(cdir, "config/secrets.yml")))
+    ENV['PATH'] = "#{ENV['PATH']}:#{tdir}"
```

Comment:
What binary are you executing in the `lib/tasks_private/terraform` dir?

Reply:
all the terraform files are in the PATH env variable.

Reply:
`PATH` is typically used to by the system to find binary files when executed by name instead of path (e.g. when you type `ls` bash knows you mean `/usr/bin/ls` because `/usr/bin` in in the `PATH` env var.

Does terraform use PATH to find its own files when not specified by a full path?

If you remove this line does this still work?